File: MonitoringSystem.cpp
#include "MonitoringSystem.hpp"
#include "Logger.hpp"
#include <iostream>

MonitoringSystem::MonitoringSystem() = default;

MonitoringSystem::~MonitoringSystem()
{
    // Очистка динамически созданных помещений.
    for (auto r : m_rooms)
        delete r;
}

void MonitoringSystem::AddRoom(Room* room)
{
    m_rooms.push_back(room);
}

void MonitoringSystem::PollSensors()
{
    std::cout << "Polling all sensors..." << std::endl;

    // Логгируем начало опроса
    Logger::GetInstance().Log("Started polling all sensors");

    for (auto r : m_rooms)
    {
        r->CheckAllSensors();

        // Логгируем успешный опрос комнаты
        Logger::GetInstance().Log("Polled room: " + r->GetName());
    }

    Logger::GetInstance().Log("Finished polling all sensors");
}

File: Lab04/Lab04Monitoring/MonitoringSystem.cs
// Класс MonitoringSystem управляет набором помещений и их датчиками.
using System.Collections.Generic;

namespace Lab04Monitoring
{
    public class MonitoringSystem
    {
        // Список всех помещений, входящих в систему мониторинга.
        private readonly List<Room> _rooms = new List<Room>();

        // Зарегистрировать новое помещение.
        public void AddRoom(Room room)
        {
            _rooms.Add(room);
        }

        // Опросить все датчики во всех помещениях.
        public void PollSensors()
        {
            Logger.Instance.Log("=== Start polling ===");

            foreach (var room in _rooms)
            {
                room.CheckAllSensors();
            }

            Logger.Instance.Log("=== End polling ===");
        }
    }
}
